### PR TITLE
issues/MCR-1398-Exception added catchEx uri resolver

### PR DESCRIPTION
**Pull request by sebhofmann**
added a catchEx URI resolver which produces xml in error case like:
<exception>
<message>$message</message>
<stacktrace>$printStackTraceOutput</stracktrace>
</exception>

### CODE REVIEW

**Comment by sebhofmann** [comment]
Please resolve https://mycore.atlassian.net/browse/MCR-1398 on merge.

**Comment by yagee-de** [comment]
Please fix commit message and open pull request against LTS branch

File: mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java [hunk 1]
```diff
@@ -23,39 +23,6 @@
...
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
-
-import javax.servlet.ServletContext;
-import javax.xml.transform.Source;
-import javax.xml.transform.TransformerException;
-import javax.xml.transform.URIResolver;
-import javax.xml.transform.sax.SAXSource;
-import javax.xml.transform.stream.StreamSource;
-
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java:58]: Keep old order of imports and change only what is necessary

**Review by yagee-de — CHANGES_REQUESTED** [review]

**Comment by sebhofmann** [comment]
Formatting should be fine now..

File: mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java [hunk 2]
```diff
@@ -1064,6 +1067,37 @@ private static long getSystemLastModified() {
...
+        public Source resolve(String href, String base) throws TransformerException {
+            String target = href.substring(href.indexOf(":") + 1);
+
+            try {
+                return MCRURIResolver.instance().resolve(target, base);
+            } catch (Exception ex) {
+                LOGGER.debug("Caught " + ex.getClass().getName() + ". Put it into XML to process in XSL!");
+                Element exception = new Element("exception");
+                Element message = new Element("message");
+                Element stacktraceElement = new Element("stacktrace");
```
Comment (yagee-de) [inline mycore-base/src/main/java/org/mycore/common/xml/MCRURIResolver.java:1082]: I think that the `<stacktrace>` element should have an attribute `xml:space="preserve"`. See [definition of xml:space](https://www.w3.org/TR/xml/#sec-white-space).

**Review by yagee-de — APPROVED** [review]

**Pull request merged**